### PR TITLE
refactor: populate url.js with urls & use in all API slices

### PR DESCRIPTION
**Pull request by Boye95**
This PR...

## Changes

- Creates a single source of truth for API endpoints/URLs thereby making the code DRY

## Fixes
-

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added/updated a component
- [ ] fixed a bug
- [ ] added/updated a new route
- [ ] added or changed css styles

Fixes #

### CODE REVIEW

File: src/app/url.js [hunk 1]
```diff
@@ -2,18 +2,120 @@
...
+    if (searchTerm && accountType) {
+      return `${url.ACCOUNTS_URL}?category=${accountType}&searchTerm=${searchTerm}&isPaginated=${isPaginated}&page=${page}&limit=${limit}`;
+    } else if (!accountType && searchTerm) {
+      return `${url.ACCOUNTS_URL}?searchTerm=${searchTerm}&isPaginated=${isPaginated}&page=${page}&limit=${limit}`;
+    } else if (accountType && !searchTerm) {
+      return `${url.ACCOUNTS_URL}?category=${accountType}&isPaginated=${isPaginated}&page=${page}&limit=${limit}`;
+    } else {
+      return `${url.ACCOUNTS_URL}?isPaginated=${isPaginated}&page=${page}&limit=${limit}`;
+    }
+  },
```
Comment (tobisupreme) [inline src/app/url.js:41]: What do you think about using ternary operators to build the URL? 

```js
GET_ALL_ACCOUNTS_URL(url, accountType, searchTerm, isPaginated, page,
  limit) {
  const categoryParam = accountType ? `category=${accountType}&` :
    '';
  const searchTermParam = searchTerm ? `searchTerm=${searchTerm}` :
    '';
  const paginatedParam = `&isPaginated=${isPaginated === true}`;
  const pageParam = (isPaginated === true && page) ?
    `&page=${page}` : '';
  const limitParam = (isPaginated === true && limit) ?
    `&limit=${limit}` : '';

  return `${url.ACCOUNTS_URL}?${categoryParam}${searchTermParam}${paginatedParam}${pageParam}${limitParam}`;
}
```

**Review by tobisupreme — CHANGES_REQUESTED** [review]

**Comment by tobisupreme** [comment]
Great job, Boye 🥳

**Review by vader-js — APPROVED** [review]

File: src/app/url.js (hunk 1 above)
Comment (Boye95) [reply src/app/url.js:41]: I actually thought about it but decided to go for readability.

Comment (vader-js) [reply src/app/url.js:41]: What I did there previously was ternary innit??

**Pull request merged**